packages/communication-react/src/mergedHooks.ts

```diff
@@ -26,10 +26,13 @@ export type Selector = (state: ClientState, props: any) => any;
 // Return undefined and skip execution when not in that context
 export const useSelector = <ParamT extends Selector | undefined>(
   selector: ParamT,
-  selectorProps?: ParamT extends Selector ? Parameters<ParamT>[1] : undefined
+  selectorProps?: ParamT extends Selector ? Parameters<ParamT>[1] : undefined,
+  type?: 'calling' | 'chat'
 ): ParamT extends Selector ? ReturnType<ParamT> : undefined => {
-  const callProps = useCallingSelector(selector as any, selectorProps);
-  const chatProps = useChatSelector(selector as any, selectorProps);
+  const callingMode = !type || type === 'calling';
+  const chatMode = !type || type === 'chat';
+  const callProps = useCallingSelector(callingMode ? (selector as any) : undefined, selectorProps);
```

Comment:
Why do we need to cast to `any` here?

Reply:
Cause selector won't always meet the type of those 2 different calling/chat selector, but we do need to walk through these 2 selectors(and no way to check their type in the runtime)